src/eFAST.jl

```diff
@@ -0,0 +1,65 @@
+Base.@kwdef struct eFAST
+    num_harmonics::Int=4
+end
+
+struct eFASTResult{T1}
+    first_order::T1
+    total_order::T1
+end
+
+function gsa(f,method::eFAST,p_range::AbstractVector,n::Int=1000;batch=false)
+    @unpack num_harmonics = method
+    num_params = length(p_range)
+    omega = [floor((n-1)/(2*num_harmonics))]
+    m = floor(omega[1]/(2*num_harmonics))
+
+    if m>= num_params-1
+        append!(omega,floor.(collect(range(1,stop=m,length=num_params-1))))
+    else
+        append!(omega,collect(range(0,stop=num_params-2)).%m .+1)
+    end
+
+    omega_temp = similar(omega)
+    first_order = []
+    total_order = []
+    s = collect((2*pi/n) * (0:n-1))
+    ps = zeros(num_params,n*num_params)
+
+    for i in 1:num_params
+        omega_temp[i] = omega[1]
+        omega_temp[[k for k in 1:num_params if k != i]] = omega[2:end]
+        l = collect((i-1)*n+1:i*n)
+        phi = 2*pi*rand()
+        for j in 1:num_params
+            x =  0.5 .+ (1/pi) .*(asin.(sin.(omega_temp[j]*s .+ phi)))
```

Comment:
`asin(sin` is just to take the mod?

Reply:
Not always? 
```
julia> asin(sin(10))
-0.5752220392306202

julia> asin(sin(pi/2))
1.5707963267948966
```

Reply:
looks like that's a mod 2pi? If not, what's it doing? It's a fairly odd piece of code.

Reply:
Oh you mean modulus, sorry, I understood it as `abs`. Yes then it's the same thing.